src/providers/namecheap.ts

```diff
@@ -0,0 +1,81 @@
+import fetch from 'node-fetch'
+import { getProviderKeys } from '../lib/data'
+import { Provider, ServiceResponse } from '../types/general'
+import { ServiceKey } from '../types/admin'
+import namecheapApi from 'namecheap-api'
+import { providerList } from '.'
+
+const provider = providerList.find(
+  provider => provider.name === 'Namecheap.com'
+)
+const { name, dns, keys, service } = provider
+
+const getKeys = (): ServiceKey[] => {
+  const keysDefault: { key: string }[] = [{ key: keys[0] }, { key: keys[1] }]
+  const providerKeys = keysDefault.map(key => {
+    const serviceKeys = getProviderKeys()
+    return serviceKeys.find(k => k.service === dns && k.key === key.key) || key
+  })
+  return providerKeys as ServiceKey[]
+}
+const findKey = (key: string): string => {
+  return (getKeys().find(k => k.key === key) || { value: '' }).value
+}
```

Comment:
could you turn `getKeys` and `findKeys` into a helper function to make the provider code simpler? Will help make future integrations simpler. `/lib/provider.ts`

Reply:
i missed the part about `/lib/provider.ts` lol but check out how i did it, and let me know what you think :)